File: 168.rb
# @param {Integer} column_number
# @return {String}
def convert_to_title(column_number)
  sheet = ''
  loop do
    column_number, r = column_number.divmod(26)
    if r == 0
      sheet += 'Z'
      column_number -= 1
    else
      sheet += (64 + r).chr
    end
    break if column_number == 0
  end
  sheet.reverse
end

p convert_to_title(1) == 'A'
p convert_to_title(28) == 'AB'
p convert_to_title(701) == 'ZY'
p convert_to_title(2147483647) == 'FXSHRXW'

File: 168.js
/**
 * @param {number} columnNumber
 * @return {string}
 */
function convertToTitle(columnNumber) {
  const divmod = (x, y) => [Math.floor(x / y), x % y];
  const nLetters = 26;
  const atSign = 64;
  let sheet = "";
  let r;
  
  while (columnNumber > 0) {
    [columnNumber, r] = divmod(columnNumber, nLetters);
    if (r === 0) {
      sheet = "Z" + sheet;
      columnNumber--;
    } else {
      sheet = String.fromCharCode(atSign + r) + sheet;
    }
  }
  return sheet;
}

console.log(convertToTitle(1) == "A");
console.log(convertToTitle(28) == "AB");
console.log(convertToTitle(701) == "ZY");
console.log(convertToTitle(2147483647) == "FXSHRXW");
